drivers/usb/udc/udc_mcux_ehci.c

```diff
@@ -0,0 +1,918 @@
+/*
+ * Copyright 2024 NXP
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+#define DT_DRV_COMPAT nxp_ehci
+
+#include <soc.h>
+#include <string.h>
+#include <stdio.h>
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/drivers/usb/udc.h>
+#include <zephyr/drivers/pinctrl.h>
+
+#include "udc_common.h"
+#include "usb.h"
+#include "usb_device_config.h"
+#include "usb_device_mcux_drv_port.h"
+#include "usb_device_ehci.h"
+#ifdef CONFIG_DT_HAS_NXP_USBPHY_ENABLED
+#include "usb_phy.h"
+#endif
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(udc_mcux, CONFIG_UDC_DRIVER_LOG_LEVEL);
+
+/*
+ * There is no real advantage to change control endpoint size
+ * but we can use it for testing UDC driver API and higher layers.
+ */
+#define USB_MCUX_MPS0		UDC_MPS0_64
+#define USB_MCUX_EP0_SIZE	64
+
+#define PRV_DATA_HANDLE(_handle) CONTAINER_OF(_handle, struct udc_mcux_data, mcux_device)
+
+struct udc_mcux_config {
+	const usb_device_controller_interface_struct_t *mcux_if;
+	void (*irq_enable_func)(const struct device *dev);
+	void (*irq_disable_func)(const struct device *dev);
+	size_t num_of_eps;
+	struct udc_ep_config *ep_cfg_in;
+	struct udc_ep_config *ep_cfg_out;
+	uintptr_t base;
+	const struct pinctrl_dev_config *pincfg;
+#ifdef CONFIG_DT_HAS_NXP_USBPHY_ENABLED
+	usb_phy_config_struct_t phy_config;
+	uint8_t need_to_init_phy;
+#endif
```

Comment:
You can simplify it to just
```c
	usb_phy_config_struct_t *phy_config;
```
Code that always builds is preferable, you also do not break multiple instance support with/without phy (no idea if it even exists.).

Reply:
Change to as only `usb_phy_config_struct_t *phy_config`